src/metabase/models/card.clj

```diff
@@ -126,43 +126,63 @@
 
 ;;; -------------------------------------------------- Hydration --------------------------------------------------
 
-(mi/define-simple-hydration-method dashboard-count
-  :dashboard_count
-  "Return the number of Dashboards this Card is in."
-  [{:keys [id]}]
-  (t2/count :model/DashboardCard, :card_id id))
-
-(mi/define-simple-hydration-method parameter-usage-count
-  :parameter_usage_count
-  "Return the number of dashboard/card filters and other widgets that use this card to populate their available
-  values (via ParameterCards)"
-  [{:keys [id]}]
-  (t2/count ParameterCard, :card_id id))
-
-(mi/define-simple-hydration-method average-query-time
-  :average_query_time
-  "Average query time of card, taken by query executions which didn't hit cache. If it's nil we don't have any query
-  executions on file."
-  [{:keys [id]}]
-  (-> (mdb.query/query {:select [:%avg.running_time]
-                        :from [:query_execution]
-                        :where [:and
-                                [:not= :running_time nil]
-                                [:not= :cache_hit true]
-                                [:= :card_id id]]})
-      first vals first))
-
-(mi/define-simple-hydration-method last-query-start
-  :last_query_start
-  "Timestamp for start of last query of this card."
-  [{:keys [id]}]
-  (-> (mdb.query/query {:select [:%max.started_at]
-                        :from [:query_execution]
-                        :where [:and
-                                [:not= :running_time nil]
-                                [:not= :cache_hit true]
-                                [:= :card_id id]]})
-      first vals first))
+(methodical/defmethod t2/batched-hydrate [:model/Card :dashboard_count]
+  [_model k cards]
+  (mi/instances-with-hydrated-data
+   cards k
+   #(->> (t2/query {:select    [[:%count.* :count] :card.id]
+                    :from      [[:report_dashboardcard :dashcard]]
+                    :left-join [[:report_card :card] [:= :dashcard.card_id :card.id]]
+                    :where     [:in :card.id (map :id cards)]
+                    :group-by  [:card.id]})
```

Comment:
Why join when you can `group by report_dashboardcard.card_id`? Seems like this will lead to smaller and simpler query.

Reply:
right, nice catch!